File: leetcode/C++/my-calendar-i.cpp
class MyCalendar {
public:
    vector<vector<int>> v;
    MyCalendar() {}
    
    bool book(int start, int end) {
        int idx=v.size();
        for(int i=0; i<v.size(); ++i){
            if(max(start,v[i][0])<min(end,v[i][1])) return false;
            if(start<v[i][0]) {idx=i; break;}
        }
        v.insert(v.begin()+idx,{start,end});
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */

File: leetcode/Python/my-calendar-i.py
class MyCalendar:

    def __init__(self):
        self.v = []

    def book(self, start: int, end: int) -> bool:
        idx = len(self.v)
        for i, (s,e) in enumerate(self.v):
            if max(s,start) < min(e,end):
                return False
            if start<s:
                idx=i
                break
        self.v.insert(idx,(start,end))
        return True


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)